openedx/features/colaraz_features/middlewares.py

```diff
@@ -0,0 +1,84 @@
+from django.conf import settings
+from django.contrib.auth import get_user, logout
+from django.http import HttpResponseRedirect
+from django.urls import reverse
+
+import third_party_auth
+from third_party_auth import pipeline
+
+from student.helpers import get_next_url_for_login_page
+from student.models import UserAttribute
+
+
+class ColarazAuthenticationMiddleware(object):
+    """
+    Special middleware for Colaraz which does following tasks:
+    1) Verify User's accessability over his requested site
+    2) Login's User if he is logged-in on Colaraz's website
+    """
+
+    def _get_user_site(self, user, session):
+        """
+        Returns domain of site associated with the User.
+        """
+        if not session.get('user_site', None):
+            session['user_site'] = UserAttribute.get_user_attribute(user, 'created_on_site')
+        return session.get('user_site', None)
+
+    def _get_request_site(self, request):
+        """
+        Returns domain of site being requested by the User.
+        """
+        site = getattr(request, "site", None)
```

Comment:
Please use single quotes for consistency.

Reply:
Done